.github/workflows/changesets-release.yml

```diff
@@ -0,0 +1,40 @@
+name: Changesets Release
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  changesets-release:
+    name: Changesets Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@main
+
+      - name: Setup Node.js
+        uses: actions/setup-node@main
+        with:
+          node-version-file: ".nvmrc"
+          cache: "pnpm"
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+
+      - name: Install Dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Build Packages
+        run: pnpm --filter "./packages/**" build
```

Comment:
My thinking here is:
1. We don't need to build anything in the `apps/` folder, since apps won't be published
2. Therefore, we don't need any environment variables such as `BIGCOMMERCE_STORE_HASH`, etc. 
3. We can speed up builds by filtering for just packages in the `packages/` folder, instead of building everything